### PR TITLE
Refactor TaskManager.FileAndForget

### PR DESCRIPTION
**Pull request by mstv**
## Proposed changes

- `TaskManager.FileAndForget()`: Remove argument `fileOnlyIf`, and replace its single usage
- Rename and refactor `TaskManager.FileAndForget()` to `RunAsyncAndForget()` in order to emphasize what it actually does,
  and save an unnecessary second call of `JoinableTaskFactory.RunAsync()` by `TaskManager.RunAsyncAndForget(asyncAction)`

In a follow-up PR, I am going to replace all `JTF.RunAsync()` and `FileAndForget()` with `RunAsyncAndForget()` (background thread) or `Control.InvokeAsync()` (UI thread), respectively.
Then I will sort the methods of `ThreadHelper` alphabetically.

## Screenshots 

N/A

## Test methodology 

- existing tests

## Please do not squash merge

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: UnitTests/GitUI.Tests/ThreadHelperTests.cs [hunk 1]
```diff
@@ -47,45 +46,6 @@ public async Task FileAndForgetIgnoresCancellationExceptions()
             Assert.Null(helper.Exception, helper.Message);
         }
 
-        [Test]
-        [Ignore("Hangs")]
```
Comment (gerhardol) [inline UnitTests/GitUI.Tests/ThreadHelperTests.cs:51]: Any point in changing this to HandleExceptions()?
(If possible enabled)

File: GitExtUtils/GitUI/TaskManager.cs [hunk 2]
```diff
@@ -67,5 +61,36 @@ public void JoinPendingOperations()
...
+        {
+            await JoinableTaskFactory.SwitchToMainThreadAsync();
+            Application.OnThreadException(ex.Demystify());
+        }
+
+        public void RunAsyncAndForget(Func<Task> asyncAction)
+        {
+            JoinableTaskFactory.RunAsync(async () =>
+                {
+                    await TaskScheduler.Default;
```
Comment (gerhardol) [inline GitExtUtils/GitUI/TaskManager.cs:75]: Why is this needed?
Is this why ConfigureAwait(false); is skipped, is the handling the same?
This seem to be the only functional change in this PR, what usage is needed?

File: GitExtUtils/GitUI/TaskManager.cs [hunk 3]
```diff
@@ -67,5 +61,36 @@ public void JoinPendingOperations()
             // since the latter adds new tasks to the collection and would therefore never complete.
             JoinableTaskContext.Factory.Run(_joinableTaskCollection.JoinTillEmptyAsync);
         }
+
+        public async Task ReportExceptionOnMainThreadAsync(Exception ex)
```
Comment (gerhardol) [inline GitExtUtils/GitUI/TaskManager.cs:65]: Why public?
Same for HandleExceptionsAsync().
Only used in RunAsyncAndForget(), kept here to show the analogy?

**Review by gerhardol — COMMENTED** [review]
This will make this code easier to understand...

File: UnitTests/GitUI.Tests/ThreadHelperTests.cs (hunk 1 above)
Comment (mstv) [reply UnitTests/GitUI.Tests/ThreadHelperTests.cs:51]: The removed testcases are duplicates of the remaining ones. They verified only that the removed argument `fileOnlyIf` worked as expected.

File: GitExtUtils/GitUI/TaskManager.cs (hunk 2 above)
Comment (mstv) [reply GitExtUtils/GitUI/TaskManager.cs:75]: > Why is this needed?
> Is this why ConfigureAwait(false); is skipped,

yes

> is the handling the same?

I expect so.

> This seem to be the only functional change in this PR, what usage is needed?

The behavior should be the same - except that `TaskManager.RunAsyncAndForget(asyncAction)` does not call `JoinableTaskFactory.RunAsync()` twice any longer.

File: GitExtUtils/GitUI/TaskManager.cs (hunk 3 above)
Comment (mstv) [reply GitExtUtils/GitUI/TaskManager.cs:65]: `public` because of generic applicability. They might be used in the follow-up PR (announced in the PR description).

Comment (RussKie) [reply GitExtUtils/GitUI/TaskManager.cs:65]: Can we please make those internal and open up only when such necessity presents itself?

File: GitExtUtils/GitUI/ThreadHelper.cs [hunk 4]
```diff
@@ -96,5 +94,11 @@ public static T CompletedResult<T>(this Task<T> task)
             return task.Result;
 #pragma warning restore VSTHRD002 // Avoid problematic synchronous waits
         }
+
+        public static void RunAsyncAndForget(Func<Task> asyncAction)
```
Comment (RussKie) [inline GitExtUtils/GitUI/ThreadHelper.cs:98]: VS threading libraries (where most of this plumbing originated from) use `FileAndForget`. Now we have two different methods - `FileAndForget` and `RunAsyncAndForget`, which is not very good.
Please keep the naming of `FileAndForget`.

File: GitExtUtils/GitUI/TaskManager.cs [hunk 5]
```diff
@@ -18,42 +18,36 @@ public TaskManager(JoinableTaskContext joinableTaskContext)
 
         public JoinableTaskFactory JoinableTaskFactory { get; init; }
 
-        public void FileAndForget(Task task, Func<Exception, bool>? fileOnlyIf = null)
+        public static async Task HandleExceptionsAsync(Func<Task> asyncAction, Func<Exception, Task> handleExceptionAsync)
```
Comment (RussKie) [inline GitExtUtils/GitUI/TaskManager.cs:21]: It'd be great if you could add appropriate xml-docs to each public method you're touching, this would help with the review as well as be helpful in the future.

File: GitExtUtils/GitUI/TaskManager.cs (hunk 2 above)
Comment (RussKie) [reply GitExtUtils/GitUI/TaskManager.cs:75]: IIRC `ConfigureAwait(false)` tells the scheduler to run continuation on any available thread. `await TaskScheduler.Default;` pushes the continuation to a background thread. So, the behaviour should be the same.

**Review by RussKie — COMMENTED** [review]
👍

**Review by gerhardol — COMMENTED** [review]
+1
Very little usage, but looks fine

**Review by RussKie — APPROVED** [review]
